File: Sprint1/C++/B/main.cpp
// https://contest.yandex.ru/contest/22449/problems/B/
#include <iostream>

using namespace std;

bool hasSameParity(int num1, int num2, int num3) {
    bool num1_is_even = num1 % 2 == 0;
    bool num2_is_even = num2 % 2 == 0;
    bool num3_is_even = num3 % 2 == 0;

    if ((num1_is_even && num2_is_even && num3_is_even) || (!num1_is_even && !num2_is_even && !num3_is_even)) {
        return true;
    }

    return false;
}

int main() {
    int num1, num2, num3;

    cin >> num1>> num2>> num3;

    if (hasSameParity(num1, num2, num3)) {
        cout << "WIN" << endl;
    } else {
        cout << "FAIL" << endl;
    }

    return 0;
}

File: Sprint1/B/Rust/src/main.rs
// https://contest.yandex.ru/contest/22449/problems/B/

use std::io;

fn has_same_parity(num1: i32, num2: i32, num3: i32) -> bool {
    let parity = num1 % 2;
    num2 % 2 == parity && num3 % 2 == parity
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();

    let numbers: Vec<i32> = input
        .split_whitespace()
        .filter_map(|n| n.parse().ok())
        .collect();

    let [num1, num2, num3] = [numbers[0], numbers[1], numbers[2]];

    println!(
        "{}",
        if has_same_parity(num1, num2, num3) {
            "WIN"
        } else {
            "FAIL"
        }
    );
}
